src/h3lib/lib/algos.c

```diff
@@ -539,12 +539,16 @@ int H3_EXPORT(hexRanges)(H3Index* h3Set, int length, int k, H3Index* out) {
 }
 
 /**
- * Returns the hollow hexagonal ring centered at origin with sides of length k.
+ * Returns the "hollow" ring of hexagons at exactly grid distance k from
+ * the origin hexagon. In particular, k=0 returns just the origin hexagon.
+ *
+ * A nonzero failure return code may happen, for example, if a pentagon is
+ * encountered. Failure cases may be fixed in future versions.
```

Comment:
If we change the handling of failure cases, would that require bumping the version to 4.0.0? I believe in that case we may have rings where sides are not the same length. (See around line 611)

Reply:
I guess this depends on which guarantees we consider to be a property of the major version. I think as @ajfriend has written it here, as long as all cells in the ring are `k` steps away from the origin, we have met the current API and no version change is needed. Adding additional failure cases is a grey area (it may cause existing code to throw errors, so may need a major bump), but removing failure cases should be fine IMO.

Reply:
I think the main user-facing change would be the need to handle the 0 values in the array, and possibly if a user made an assumption that `hexRing` always returns a hexagon shape.

Reply:
Handling 0 values in the array is a good point. Do we want to mention that users currently **do not** need to filter out zeros? Or do we want to say that they should always be wary?

Reply:
I'd suggest maybe leaving it a little vague for what happens in the cases that currently fail. That would allow us to address this when we have a clearer idea of what the new behavior is.

Reply:
sounds good. I'll land after making the change suggested by @nrabinowitz